lib/travis/cli/encrypt_file.rb

```diff
@@ -70,14 +70,14 @@ def decrypt_command(path)
         "openssl aes-256-cbc -K $#{env_name(:key)} -iv $#{env_name(:iv)} -in #{escape_path(path)} -out #{escape_path(decrypt_to)} -d"
       end
 
-      def set_env_vars
+      def set_env_vars(input_path)
         say "storing secure env variables for decryption"
-        repository.env_vars.upsert env_name(:key), key, :public => false
-        repository.env_vars.upsert env_name(:iv),  iv,  :public => false
+        repository.env_vars.upsert env_name(input_path, :key), key, :public => false
+        repository.env_vars.upsert env_name(input_path, :iv),  iv,  :public => false
       end
 
-      def env_name(name)
-        @env_prefix ||= "encrypted_#{Digest.hexencode(Digest::SHA1.digest(Dir.pwd)[0..5])}"
+      def env_name(input_path, name)
```

Comment:
Tiny caveat here is that, as it was before, this command is context dependent. If we execute `travis encrypt-file` from a different directory, even if the command is invoked on the same file, we will compute a different hash. This is not documented anywhere, but it is worth pointing it out.

Reply:
@BanzaiMan without having this verified, i believe that `input_path` is whatever path is passed to `travis encrypt path/to/file` (i.e. it would be `path/to/file`). so yes, i guess that counts as context dependent, but it kinda makes sense to me, as such paths would be considered unique per repo, too (and so would the resulting env vars)?